modules/@apostrophecms/piece-type/index.js

```diff
@@ -350,21 +351,34 @@ module.exports = {
       },
       composeFilters() {
         self.filters = options.filters || [];
-        if (options.addFilters) {
+        if (Array.isArray(options.addFilters)) {
           _.each(options.addFilters, function (newFilter) {
             // remove it from the filters if we've already added it, last one wins
             self.filters = _.filter(self.filters, function (filter) {
               return filter.name !== newFilter.name;
             });
-            // add the new field to the filters
-            self.filters.push(newFilter);
           });
+          // add the new field to the filters
+          self.filters = options.addFilters.concat(self.filters);
```

Comment:
This may be controversial. It assumes the project isn't adding things to `addFilters` and that `addFilters` is the system filters, as it is here. I'm aiming to get Published and Trash at the front of the array. This may be a reflection of the need to update this module.

Reply:
...

[Lots of typing elided]

Right now, only schema `fields` get a special module format section. We don't want to have a whole slew of module sections with their own special rules, it gets unwieldy quick.

But I find this answer unsatisfying too. I'd like you to move ahead with this ticket sticking to the current way `addFilters` works, but I will work on a proposal and a discussion to solve it better.